include/flamegpu/gpu/CUDASimulation.h

```diff
@@ -247,14 +253,18 @@ class CUDASimulation : public Simulation {
      */
     std::vector<double> getElapsedTimeSteps() const;
 
-
     /** 
      * Get the duration of an individual step in seconds.
      * @param step Index of step, must be less than the number of steps executed.
      * @return elapsed time of required step in seconds
      */
     double getElapsedTimeStep(unsigned int step) const;
 
+    /**
+     * Determines which agents require sorting - only used once during initialisation. Must be called manually if not using .simulate()
+     */
+    void determineAgentsToSort();
```

Comment:
I really think this should be hidden from the user. Not looked at it's implementation yet, but I presume it can be built into the initialisation of `CUDASimulation`, regardless of calls to `simulate()`.

Reply:
Yeah, having looked at this, i don't see why it can't simply be at the end of the constructor or similar.